File: Data Structures & Algorithms/count-paths/submission-0.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>> dp(m, vector<int>(n, 0));
        
        int r = 0, c = 0;
        for (r = 0; r < m; r++){
            dp[r][c] = 1;
        }
        
        r = 0;
        for (c = 0; c < n; c++){
            dp[r][c] = 1;
        }

        for (r = 1; r < m; r++){
            for(c = 1; c < n; c++){
                dp[r][c] = dp[r - 1][c] + dp[r][c - 1];
            }
        }
        

        return dp[m-1][n-1];
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ret = 0
        curr = 0
        prefix = {0 : 1}

        for num in nums:
            curr += num
            diff = curr - k
            ret += prefix.get(diff, 0)
            prefix[curr] = 1 + prefix.get(curr, 0)
        
        return ret